metrics/web/test/features/auth/domain/usercases/test_utils/error_user_repository_mock.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:metrics/features/auth/domain/entities/authentication_exception.dart';
+import 'package:metrics/features/auth/domain/repositories/user_repository.dart';
+import 'package:mockito/mockito.dart';
+
+/// Mock implementation of the [UserRepository] that throws
+/// an [AuthenticationException] from all methods.
+class ErrorUserRepositoryMock extends Mock implements UserRepository {
+  static const authException = AuthenticationException();
+
+  ErrorUserRepositoryMock() {
+    when(authenticationStream()).thenThrow(authException);
+    when(signInWithEmailAndPassword(any, any)).thenThrow(authException);
+    when(signOut()).thenThrow(authException);
```

Comment:
I think this should be placed in tests since it provides unobvious dependencies between tests where this mock is used. From the [Best Practices section](https://pub.dev/packages/mockito#best-practices) of the mockito package: 
>A class which extends Mock should never stub out it's own responses with when in it's constructor or anywhere else. Stubbed responses should be defined in the tests where they are used.